### ISSUE TITLE
rewrite fzopen plugin

### ISSUE THREAD

**Issue by akimdi** [jarun/nnn#522]
#### Environment details (Put `x` in the checkbox along with the information)

[x] Operating System: ArchLinux
[x] Desktop Environment: i3wm
[x] Terminal Emulator: kitty
[x] Shell: zsh 5.8
[ ] Custom desktop opener (if applicable):
[ ] Issue exists on `nnn` master

Is it possible to rewrite the plugin for a more comprehensive search?
Skim contains the same as fzf. In addition it has several improvements.
Mainly difference fuzzy search algorithm.
something like this `xdg-open $(sk -m)`
https://github.com/lotabout/skim#difference-to-fzf

**Comment by jarun** [comment]
@KlzXS please check.

**Comment by jarun** [comment]
Are you suggesting we use skim instead of fzf?

**Comment by KlzXS** [comment]
I think we can add skim to this plugin. I don't know about adding it as a default.
What is the main selling point, the multi-select feature or is it the interactive mode? From what I've gathered that is the only difference besides the "different" algorithm.

**Comment by akimdi** [comment]
Yes. Skim instead of fzf.
Skim includes everything that is in fzf.
Аnd additionally opens up other possibilities.

skim can search by folders and files
fzf searches by only file name

for example
https://i.imgur.com/klFsgsX.png

simple use case
You need to find the info file
there are several such files of the same name, they are different and are located in different folders, but they are called the same.
you forgot the name of the file itself, but remember that it was in folder temp
skim will help here
simply put, this is the difference between fuzzy search skim and fzf
https://youtu.be/iLTlQ2aTUB0

in the current plugin it is done like this
https://github.com/jarun/nnn/blob/c63fb28effb2fbae903928b9b9453974c6e96c9f/plugins/fzopen#L28

I tried to do something like
```xdg-open $ (sk -m)```

it works, but not always
because I don’t know how to escape spaces and other characters

**Comment by akimdi** [comment]
Yes.
multi-select feature and interactive mode is a bonus

**Comment by jarun** [comment]
> Skim instead of fzf.

fzf is way more popular than I thought myself. We actually supported only fzy earlier. And then I noticed fzy author doesn't respond to defects on GitHub. And everyone uses fzf.

You can have skim as fallback but not the default.

> because I don’t know how to escape spaces and other characters

You have to invoke xdg-open which takes only one file. That's what fzopen does, it opens the file. How you'll do is up to you. :)

**Comment by akimdi** [comment]
> You can have skim as fallback but not the default.
> 
> >

maybe can make a plugin ```skopen```?



> You have to invoke `xdg-open` which takes only one file.  

no other options to open multiple files?



> That's what fzopen does, it opens the file.

yes, but the plugin fzopen uses inside fzf and it has a fuzzy search only by file names.
I wrote above that it’s not always possible to find what you want.
https://youtu.be/iLTlQ2aTUB0

**Comment by jarun** [comment]
> maybe can make a plugin skopen?

No, please.

> no other options to open multiple files?

Plugins are indicative for personal modification. So users can change it.

@KlzXS please take a look. Beyond me.

**Comment by akimdi** [comment]
It's a pity.
Thank you.

**Comment by jarun** [comment]
I don't deal with plugins everyday. So I requested @KlzXS to check it. At the same time, we can't make skim the default. You'll have to accommodate it in the existing plugin.

**Comment by KlzXS** [comment]
I added support for skim in a PR. We can't invoke it with `-m` as passing multiple files to xdg-open would give an argument error.

But just as you had problems with fzf no finding the file you are looking for I had problems with finding the file I was looking for with skim. There is no one size fits all solution.

**Comment by akimdi** [comment]
@KlzXS   how to use skim in fzopen plugin?
I updated all the plugins
fzf still works for me

**Comment by leovilok** [comment]
@akimdi the default will stay `fzf`, if it is installed. To use `skim` instead, either:
1. uninstall `fzf` and `fzy` if you don't use them or
2. edit `~/.config/nnn/plugins/fzopen`, and make `skim` the default

**Comment by akimdi** [comment]
@jarun  
@leovilok 
@KlzXS 

Amazing!
Thank you very much!
